sentinelhub/areas.py

```diff
@@ -475,3 +484,152 @@ def _make_split(self):
 
                         self.bbox_list.append(bbox)
                         self.info_list.append(info)
+
+
+class BaseUtmSplitter(AreaSplitter):
+    """ Base splitter that returns bboxes of fixed size aligned to UTM zones or UTM grid tiles as defined by the MGRS
+    """
+    def __init__(self, shape_list, crs, bbox_size):
+        """
+        :param shape_list: A list of geometrical shapes describing the area of interest
+        :type shape_list: list(shapely.geometry.multipolygon.MultiPolygon or shapely.geometry.polygon.Polygon)
+        :param crs: Coordinate reference system of the shapes in `shape_list`
+        :type crs: CRS
+        :param bbox_size: Physical size in metres of generated bounding boxes. Could be a float or tuple of floats
+        :type bbox_size: int or (int, int) or float or (float, float)
+        """
+        super().__init__(shape_list, crs)
+
+        self.bbox_size = self._parse_split_parameters(bbox_size, allow_float=True)
+
+        self.shape_geometry = Geometry(self.area_shape, self.crs).transform(CRS.WGS84)
+
+        self.utm_grid = self._get_utm_polys()
+
+        self._make_split()
+
+    @abstractmethod
+    def _get_utm_polys(self):
+        raise NotImplementedError
+
+    @staticmethod
+    def _get_utm_from_props(utm_dict):
+        """ Return the UTM CRS corresponding to the UTM described by the properties dictionary
+
+        :param utm_dict: Dictionary reporting name of the UTM zone and MGRS grid
+        :type utm_dict: dict
+        :return: UTM coordinate reference system
+        :rtype: sentinelhub.CRS
+        """
+        return CRS('32{}{}'.format(6 if utm_dict['direction'] == 'N' else 7, str(utm_dict['zone']).zfill(2)))
+
+    def _align_bbox_to_size(self, bbox):
+        """ Align input bbox coordinates to be multiples of the bbox size
+
+        :param bbox: Bounding box in UTM coordinates
+        :type bbox: sentinelhub.BBox
+        :return: BBox objects with coordinates multiples of the bbox size
+        :rtype: sentinelhub.BBox
+        """
+        size_x, size_y = self.bbox_size
+        lower_left_x, lower_left_y = bbox.get_lower_left()
+        return BBox([(lower_left_x - lower_left_x % size_x, lower_left_y - lower_left_y % size_y),
```

Comment:
It is numerically more precise to do 
```Python
math.floor(lower_left_x / size_x) * size_x
```
instead of
```Python
lower_left_x - lower_left_x % size_x
```
Even though % operation works for floats I would avoid using it.

Reply:
changed